streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java

```diff
@@ -191,7 +191,7 @@
      *   the instance will be in the ERROR state. The user will need to close it.
      */
     public enum State {
-        CREATED(1, 3), REBALANCING(2, 3, 5), RUNNING(1, 3, 5), PENDING_SHUTDOWN(4), NOT_RUNNING, ERROR(3, 5);
+        CREATED(1, 3), REBALANCING(2, 3, 5), RUNNING(1, 3, 5), PENDING_SHUTDOWN(4), NOT_RUNNING, ERROR(3);
```

Comment:
This is a minor cleanup piggy-backed in this PR: since in `setState` we already exclude `ERROR` to transit to itself so that the user's registered listener is triggered multiple times necessarily, we can actually exclude this transition all together in the diagram.